docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,555 @@
+
+### Final Code (Function)
+
+Below is the complete code snippet of this guide, with all the steps compiled together.
+
+```js
+import { ethers } from 'ethers';
+import { ERC725YDataKeys } from '@lukso/lsp-smart-contracts';
+
+import { abi as UniversalProfileABI } from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
+
+import { abi as TipMeABI } from './TipMe.json';
+import { bytecode as TipMeBytecode } from './TipMe.json';
+
+async function main() {
+  // RPC URL (e.g., LUKSO testnet)
+  const RPC_URL = 'https://rpc.testnet.lukso.network';
+
+  // Make sure the address associated with this private key has the 'ADDEXTENSIONS' and 'CHANGEEXTENSIONS' permissions
+
+  // Replace with your private key
+  const privateKey = 'your-private-key';
+
+  const provider = new ethers.providers.JsonRpcProvider(RPC_URL);
+  const signer = new ethers.Wallet(privateKey, provider);
+
+  const universalProfileAddress = 'your-universal-profile-address';
+
+  const universalProfile = new ethers.Contract(
+    universalProfileAddress,
+    UniversalProfileABI,
+    signer,
+  );
+
+  const TipMeFactory = new ethers.ContractFactory(
+    TipMeABI,
+    TipMeBytecode,
+    signer,
+  );
+
+  const tipMeContract = await TipMeFactory.deploy();
+  await tipMeContract.deployTransaction.wait();
+
+  const tipMeInterface = new ethers.utils.Interface(TipMeABI);
+  const tipMeFunctionSelector = tipMeInterface.getSighash('tipMe()');
+
+  // Define the DataKey for the extension
+  const extensionDataKey =
+    ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
+    tipMeFunctionSelector.substr(2) +
+    '0'.repeat(32);
+
+  const tx = await universalProfile.setData(
+    extensionDataKey,
+    tipMeContract.address,
+  );
+  await tx.wait();
+
+  // Attaching the ABI of TipMe to the Universal Profile so we can call tipMe() function directly on it
+  const universalProfileWithExtension = new ethers.Contract(
+    universalProfileAddress,
+    TipMeABI,
+    signer,
+  );
+
+  // Calling the tipMe() function on the Universal Profile
+  const tipMeTx = await universalProfileWithExtension.tipMe({
+    value: ethers.utils.parseEther('0.1'),
+  });
+  await tipMeTx.wait();
+}
+
+main();
+```
+
+## Extending InterfaceIds
+
+Extending interfaceIds is crucial for Universal Profiles, as many protocols check for support of specific interfaceIds before executing calls to certain functions. Extending interfaceIds typically follows a similar approach to extending functions, with the main distinction being the existance of the `supportsInterface` function.
+
+### Step 1: Create a Contract Supporting a Specific InterfaceId
+
+First, you need to create a Solidity contract that implements the supportsInterface function. This function should return true for at least one specific interfaceId, in this case, a dummy interfaceId like `0xaabbccdd`.
```

Comment:
```suggestion
First, you need to create a Solidity contract that implements the `supportsInterface` function. This function should return true for at least one specific interfaceId, in this case, a dummy interfaceId like `0xaabbccdd`.
```

adding links and reference to supportsInterface (for instance like this one https://docs.lukso.tech/contracts/contracts/ERC725/#supportsinterface ) can't hurt imo